R/geom-.R

```diff
@@ -208,6 +211,9 @@ Geom <- ggproto("Geom",
   extra_params = c("na.rm"),
 
   parameters = function(self, extra = FALSE) {
+    if (!is.null(self$default_params)) {
+      return(names(self$default_params))
+    }
```

Comment:
2) We simply return the name of the default parameter list for checking `layer()` input. For backward compatibility, if this list isn't present, we resort to the old way of determining input.